deploy/runtools/run_farm.py

```diff
@@ -489,59 +504,59 @@ def unload_xdma(self):
         #self.instance_logger("Waiting 10 seconds after removing kernel modules (esp. xocl).")
         #time.sleep(10)
 
-    def clear_fpgas(self):
+    def clear_fpgas(self) -> None:
         # we always clear ALL fpga slots
-        for slotno in range(self.parentnode.get_num_fpga_slots_max()):
-            self.instance_logger("""Clearing FPGA Slot {}.""".format(slotno))
-            with StreamLogger('stdout'), StreamLogger('stderr'):
-                remote_kmsg("""about_to_clear_fpga{}""".format(slotno))
-                run("""sudo fpga-clear-local-image -S {} -A""".format(slotno))
-                remote_kmsg("""done_clearing_fpga{}""".format(slotno))
+        if isinstance(self.parentnode, F1_Instance):
+            for slotno in range(self.parentnode.get_num_fpga_slots_max()):
+                self.instance_logger("""Clearing FPGA Slot {}.""".format(slotno))
+                with StreamLogger('stdout'), StreamLogger('stderr'):
+                    remote_kmsg("""about_to_clear_fpga{}""".format(slotno))
+                    run("""sudo fpga-clear-local-image -S {} -A""".format(slotno))
+                    remote_kmsg("""done_clearing_fpga{}""".format(slotno))
 
-        for slotno in range(self.parentnode.get_num_fpga_slots_max()):
-            self.instance_logger("""Checking for Cleared FPGA Slot {}.""".format(slotno))
-            with StreamLogger('stdout'), StreamLogger('stderr'):
-                remote_kmsg("""about_to_check_clear_fpga{}""".format(slotno))
-                run("""until sudo fpga-describe-local-image -S {} -R -H | grep -q "cleared"; do  sleep 1;  done""".format(slotno))
-                remote_kmsg("""done_checking_clear_fpga{}""".format(slotno))
+            for slotno in range(self.parentnode.get_num_fpga_slots_max()):
+                self.instance_logger("""Checking for Cleared FPGA Slot {}.""".format(slotno))
+                with StreamLogger('stdout'), StreamLogger('stderr'):
+                    remote_kmsg("""about_to_check_clear_fpga{}""".format(slotno))
+                    run("""until sudo fpga-describe-local-image -S {} -R -H | grep -q "cleared"; do  sleep 1;  done""".format(slotno))
+                    remote_kmsg("""done_checking_clear_fpga{}""".format(slotno))
```

Comment:
How does this look once you have 'what-to-run' implemented?  I strongly dislike the current code silently does nothing `unless isinstance(self.parentnode, F1_Instance)` if for no other reason than we no longer "always clear ALL fpga slots" when `clear_fpgas` is called.  Should probably make this a `match` statement with default case that throws `NotImplemented` so that we don't open the door for really difficult to understand silent failure cases.

However, if 'where-to-run' takes care of this by abstracting this method behind a dispatcher that has to implement this method, I'm willing to look the other way here.

Reply:
These if's are really no-ops (with how things are currently used). In reality, these functions are called properly in `infrasetup` (https://github.com/firesim/firesim/blob/043905e14ddb304705226738df14dc8885faf436/deploy/runtools/run_farm.py#L676)(or other types of functions) with an `if` in that function doing the same thing. If the preference is adding an assert instead of silently skipping that's fine with me. I was just trying to avoid adding typing changes that affect the original code too much.

Reply:
@timsnyder-siv is this the only issue with the PR? If so I'll address this and I think we can merge this (since it doesn't change any functionality)

Reply:
Solved offline. We will revert to using `assert`s